### DISCUSSION TITLE
Adding more track and album properties?

### DISCUSSION THREAD

**Discussion by ali-ramadhan** [MoeMusic/Moe#282] [Ideas]
What is the current thinking surrounding adding new properties to the `Album` and `Track` classes? I know they both have a dictionary for `custom` fields so maybe no additional fields are needed. Should, for example, the Musicbrainz plugin be making use of the `custom` dict?

Two examples I'm thinking of to start:
1. An `album.composer` property would help with tagging classical music, but would usually be empty for other genres. Or should it be `track.composer`...?
2. An `album.type` property can differentiate albums by one artist, compilations, and soundtracks. I believe Musicbrainz can provide this data. It can help with tagging compilations and soundtracks differently.

A third example which is a bit different. Right now `track.genres` is a property but an album has no genre. Does it make sense to add an `album.genres` field? Should it be the intersection of all track genres? Or could the album genre be different from the genre of the tracks? From a music tagging perspective I'm not sure.

Sorry if I'm just rambling here!

**Comment by jtpavlock** [comment]
> What is the current thinking surrounding adding new properties to the Album and Track classes? I know they both have a dictionary for custom fields so maybe no additional fields are needed. Should, for example, the Musicbrainz plugin be making use of the custom dict?

The only properties that I believe don't belong as native properties in Albums and Tracks are those specific to particular platforms e.g. musicbrainz. Everything else is kind of subjective, but typically based on existing support for the tag or field in question. Musicbrainz actually does utilize the custom dict for tracks and albums, right now it just sets the `mb_album_id` and `mb_track_id` fields within the custom dict. 

> An album.composer property would help with tagging classical music, but would usually be empty for other genres. Or should it be track.composer...?

I think composer is a good example of a tag to add. It's got well documented support by various media players, and our underlying tagging software, [mediafile](https://github.com/beetbox/mediafile), also supports it. I also think it should be track.composer instead of album.composer, as I don't think there's any distinction between an album_composer and a track_composer in tags like there is between "albumartist" and just "artist" (although please correct me if I'm wrong).

If you'd like to see this implemented, please feel free to open a PR or even just a feature request.

> An album.type property can differentiate albums by one artist, compilations, and soundtracks. I believe Musicbrainz can provide this data. It can help with tagging compilations and soundtracks differently.

I'd have to do some more digging into this field to see if it's supported by mediafile or if the information is standardized at all. If it's only specific to Musicbrainz, I'd say it should be a custom musicbrainz field instead like `mb_release_type` or something.

> A third example which is a bit different. Right now track.genres is a property but an album has no genre. Does it make sense to add an album.genres field? Should it be the intersection of all track genres? Or could the album genre be different from the genre of the tracks? From a music tagging perspective I'm not sure.

This is tricky, as there's tags for a track's "genre", but no tag like "albumgenre". I'm not sure how useful an album_genre would be if it can't be used or seen by other software, and I'm not sure how many players or databases even distinguish between an album's genre and a track's genre. If there's a good use case for it, I would be open to adding it.
